File: Client/src/app/services/loss-type.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { PaginatedResult } from '../models/paginated-result.model';
import { LossType } from '../models/loss-type.model';
import { environment } from '../../environments/environment';
import { Observable, of } from 'rxjs';
import { catchError } from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class LossTypeService {

  constructor(private readonly http: HttpClient) { }

  public getPage(pageSize: number, pageIndex: string): Observable<PaginatedResult<LossType>> {
  		return this.http.get<PaginatedResult<LossType>>(environment.apiUrl + `/api/LossType?pageSize=${pageSize}&pageIndex=${pageIndex}`)
  			.pipe(
  				catchError(() => of(new PaginatedResult<LossType>()))
			);
  	}
}

File: ClaimsHandling.Domain/Services/LossTypeService.cs
using ClaimsHandling.Domain.Interfaces;
using ClaimsHandling.Domain.Models;

namespace ClaimsHandling.Domain.Services
{
    public class LossTypeService : ILossTypeService
    {
        private readonly ILossTypeRepository _lossTypeRepository;

        public LossTypeService(ILossTypeRepository lossTypeRepository)
        {
            _lossTypeRepository = lossTypeRepository;
        }

        public PaginatedResult<LossType> GetPage(int pageSize,
                                                 int pageIndex)
        {
            return _lossTypeRepository.GetPage(pageSize, pageIndex);
        }
    }
}
